src/main.rs

```diff
@@ -48,13 +48,18 @@ struct App {
     ///
     /// This disables all online audit rules, and prevents zizmor from
     /// auditing remote repositories.
-    #[arg(short, long, env = "ZIZMOR_OFFLINE", group = "_offline")]
+    #[arg(short, long, env = "ZIZMOR_OFFLINE", 
+        conflicts_with_all = ["gh_token", "gh_hostname"])]
     offline: bool,
 
     /// The GitHub API token to use.
-    #[arg(long, env, group = "_offline")]
+    #[arg(long, env)]
     gh_token: Option<String>,
 
+    /// The GitHub API Server Hostname, if you are running GHE.
+    #[arg(long, env)]
+    gh_hostname: Option<String>,
```

Comment:
Rather than a `None` state, perhaps we can default to `https://api.github.com/` here? That way there's no need for the subsequent `Option<...>` handling 🙂

Reply:
We could, but there should be handing at github_api.rs making api_base,
because we have to add /api/v3 if it is on GHE server..

Should i change it to String?

Reply:
> because we have to add /api/v3 if it is on GHE server..

I might be missing something, but is this always true? The GHE docs don't seem to consistently mention an `/api/v3` prefix for API routes. I'm wondering if it's a redundant prefix, i.e. whether `/{route} and `/api/v3/{route}` both work.

Reply:
https://docs.github.com/en/enterprise-server@3.15/rest/quickstart?apiVersion=2022-11-28&tool=curl

Api docs for ghe server mentions /api/v3,
but I'll test it again to make sure!

Reply:
Ah yep, looks like there's no redirect, at least for `github.com`.



> (Maybe gh handles this internally..?)

Yep, it looks like they do:

https://github.com/cli/cli/blob/6c5145166003ac6fb952c5c591a6f3bdeea10465/internal/ghinstance/host.go#L59-L70

That uses `ghauth.IsEnterprise`, which seems to come from here:

https://github.com/cli/go-gh/blob/65bd8d766abd062846dd5cffe21defab7c0fe4c5/pkg/auth/auth.go#L159-L166

So IMO we should copy that logic :slightly_smiling_face: